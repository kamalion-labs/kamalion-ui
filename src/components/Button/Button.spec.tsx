import React from 'react';
import renderer from 'react-test-renderer';
//import { render, screen } from '@testing-library/react';
import { Button } from '.';
import { FaBeer } from 'react-icons/fa';

describe('Button', () => {
  it('should render primary Button', () => {
    const button = renderer.create(<Button type="primary">Primary</Button>).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should render secondary Button', () => {
    const button = renderer.create(<Button type="secondary">Secondary</Button>).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should render danger Button', () => {
    const button = renderer.create(<Button type="danger">Danger</Button>).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should render warning Button', () => {
    const button = renderer.create(<Button type="warn">Warning</Button>).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should render info Button', () => {
    const button = renderer.create(<Button type="info">Info</Button>).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should render success Button', () => {
    const button = renderer.create(<Button type="success">Success</Button>).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should render white Button', () => {
    const button = renderer.create(<Button type="white">White</Button>).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should render submit Button', () => {
    const button = renderer.create(<Button submit>Submit</Button>).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should have className', () => {
    const button = renderer.create(<Button className="bg-white">Classname</Button>).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should have icon only', () => {
    const button = renderer.create(<Button icon={<FaBeer />} />).toJSON();

    expect(button).toMatchSnapshot();
  });

  // it('should be clickable', () => {
  //   const mockCallBack = jest.fn();

  //   render(
  //     <Button type="primary" onClick={mockCallBack}>
  //       Button
  //     </Button>
  //   );
  //   // button.simulate('click');
  //   // expect(mockCallBack).toBeCalledTimes(1);
  //   //expect(button).toMatchSnapshot();
  // });

  // it('should have icon and be clickable', () => {
  //   const mockCallBack = jest.fn();

  //   render(
  //     <Button type="primary" icon={<FaBeer />} onClick={mockCallBack}>
  //       Button
  //     </Button>
  //   );

  //   const button = screen.getBy

  //   button.simulate('click');
  //   expect(mockCallBack).toBeCalledTimes(1);
  //   expect(button).toMatchSnapshot();
  // });

  // it('should have loader and be clickable', () => {
  //   const mockCallBack = jest.fn();

  //   const button = shallow(
  //     <Button type="primary" onClick={mockCallBack} usesLoader loadingText="Loading...">
  //       Button with loader
  //     </Button>
  //   );
  //   button.simulate('click');
  //   expect(mockCallBack).toBeCalledTimes(1);

  //   button.update();
  //   expect(button).toMatchSnapshot();
  // });

  // it('should have loader and icon and be clickable', () => {
  //   const mockCallBack = jest.fn();

  //   const button = shallow(
  //     <Button type="primary" onClick={mockCallBack} icon={<FaBeer />} usesLoader loadingText="Loading...">
  //       Button with loader
  //     </Button>
  //   );
  //   button.simulate('click');
  //   expect(mockCallBack).toBeCalledTimes(1);

  //   button.update();
  //   expect(button.find('.icon-spin')).toHaveLength(1);
  //   expect(button).toMatchSnapshot();
  // });

  // it('should have icon only and a loader and be clickable', () => {
  //   const mockCallBack = jest.fn();

  //   const button = shallow(<Button icon={<FaBeer />} usesLoader onClick={mockCallBack} />);
  //   button.simulate('click');
  //   expect(mockCallBack).toBeCalledTimes(1);

  //   button.update();
  //   expect(button.find('.icon-spin')).toHaveLength(1);
  //   expect(button).toMatchSnapshot();
  // });
});
