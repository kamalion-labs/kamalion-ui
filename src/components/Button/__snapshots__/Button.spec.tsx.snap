// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should have className 1`] = `
<button
  className="button button-primary bg-white"
  onClick={[Function]}
  type="button"
>
  <div
    className="flex flex-row h-full"
  >
    <div
      className="px-5 py-1 items-center flex text-center w-full"
    >
      Classname
    </div>
  </div>
</button>
`;

exports[`Button should have icon only 1`] = `
<button
  className="button button-primary "
  onClick={[Function]}
  type="button"
>
  <div
    className="flex flex-row h-full"
  >
    <div
      className="button-icon rounded-r-md"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M368 96h-48V56c0-13.255-10.745-24-24-24H24C10.745 32 0 42.745 0 56v400c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24v-42.11l80.606-35.977C429.396 365.063 448 336.388 448 304.86V176c0-44.112-35.888-80-80-80zm16 208.86a16.018 16.018 0 0 1-9.479 14.611L320 343.805V160h48c8.822 0 16 7.178 16 16v128.86zM208 384c-8.836 0-16-7.164-16-16V144c0-8.836 7.164-16 16-16s16 7.164 16 16v224c0 8.836-7.164 16-16 16zm-96 0c-8.836 0-16-7.164-16-16V144c0-8.836 7.164-16 16-16s16 7.164 16 16v224c0 8.836-7.164 16-16 16z"
        />
      </svg>
    </div>
  </div>
</button>
`;

exports[`Button should render danger Button 1`] = `
<button
  className="button button-danger "
  onClick={[Function]}
  type="button"
>
  <div
    className="flex flex-row h-full"
  >
    <div
      className="px-5 py-1 items-center flex text-center w-full"
    >
      Danger
    </div>
  </div>
</button>
`;

exports[`Button should render info Button 1`] = `
<button
  className="button button-info "
  onClick={[Function]}
  type="button"
>
  <div
    className="flex flex-row h-full"
  >
    <div
      className="px-5 py-1 items-center flex text-center w-full"
    >
      Info
    </div>
  </div>
</button>
`;

exports[`Button should render primary Button 1`] = `
<button
  className="button button-primary "
  onClick={[Function]}
  type="button"
>
  <div
    className="flex flex-row h-full"
  >
    <div
      className="px-5 py-1 items-center flex text-center w-full"
    >
      Primary
    </div>
  </div>
</button>
`;

exports[`Button should render secondary Button 1`] = `
<button
  className="button button-secondary "
  onClick={[Function]}
  type="button"
>
  <div
    className="flex flex-row h-full"
  >
    <div
      className="px-5 py-1 items-center flex text-center w-full"
    >
      Secondary
    </div>
  </div>
</button>
`;

exports[`Button should render submit Button 1`] = `
<button
  className="button button-primary "
  onClick={[Function]}
  type="submit"
>
  <div
    className="flex flex-row h-full"
  >
    <div
      className="px-5 py-1 items-center flex text-center w-full"
    >
      Submit
    </div>
  </div>
</button>
`;

exports[`Button should render success Button 1`] = `
<button
  className="button button-success "
  onClick={[Function]}
  type="button"
>
  <div
    className="flex flex-row h-full"
  >
    <div
      className="px-5 py-1 items-center flex text-center w-full"
    >
      Success
    </div>
  </div>
</button>
`;

exports[`Button should render warning Button 1`] = `
<button
  className="button button-warn "
  onClick={[Function]}
  type="button"
>
  <div
    className="flex flex-row h-full"
  >
    <div
      className="px-5 py-1 items-center flex text-center w-full"
    >
      Warning
    </div>
  </div>
</button>
`;

exports[`Button should render white Button 1`] = `
<button
  className="button button-white "
  onClick={[Function]}
  type="button"
>
  <div
    className="flex flex-row h-full"
  >
    <div
      className="px-5 py-1 items-center flex text-center w-full"
    >
      White
    </div>
  </div>
</button>
`;
